src/Avalonia.Controls/CalendarDatePicker/CalendarDatePicker.cs

```diff
@@ -730,7 +730,9 @@ private void OpenDropDown()
             // the TextParseError event
             try
             {
-                newSelectedDate = DateTime.Parse(text, DateTimeHelper.GetCurrentDateFormat());
+                newSelectedDate = !string.IsNullOrEmpty(this.CustomDateFormatString) ?
```

Comment:
The condition should also check that `SelectedDateFormat == Custom`

Reply:
@MrJul done